xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -837,6 +837,78 @@ void CActiveAESink::OpenSink()
   CLog::Log(LOGDEBUG, "CActiveAESink::OpenSink - trying to open device {}", device);
   m_sink = CAESinkFactory::Create(device, m_sinkFormat);
 
+  // try other device with same driver and same friendly name
+  // is probably the same device renamed due driver update
+  if (!m_sink && !m_sinkInfoList.empty() && !dev.driver.empty() && !dev.friendlyName.empty())
+  {
+    for (const auto& sink : m_sinkInfoList)
+    {
+      if (sink.m_sinkName != dev.driver)
+        continue;
+
+      if (m_sink)
```

Comment:
When could this be happen? The CAESinkFactory::Create is followed by a break already.

Reply:
There are two nested loops and it takes two breaks to exit, but it is true that it is better to put it where you say.